src/Core/ServiceWrapper/Main.cs

```diff
@@ -935,6 +1033,34 @@ private static void InitLoggers(ServiceDescriptor d, bool enableCLILogging)
                 appenders.ToArray());
         }
 
+        private static unsafe bool IsProcessElevated()

```

Comment:
I wish there was a safe version provided by .NET API. It is ok as a workaround

Reply:
FTR there is another version.

```cs
using (WindowsIdentity identity = WindowsIdentity.GetCurrent())
{
    WindowsPrincipal principal = new WindowsPrincipal(identity);
    if (!principal.IsInRole(WindowsBuiltInRole.Administrator))
    {
        return;
    }
}
```

Reply:
What is downside of this another option? AFAICT it is supported even in .NET Framework 2.0

Reply:
They are different. An elevated token isn't necessarily in the Administrators group.

> AFAICT it is supported even in .NET Framework 2.0

It reminds me that UAC was not invented on Windows XP. Do we still support Windows XP?

Reply:
I definitely know some Jenkins users who still run Windows XP. Whether we want to support it or not, it is a separate question. I am happy to officially drop WinXP support once it becomes an obstacle.

Created https://github.com/winsw/winsw/issues/497 for Windows support policy documentation